File: 2844-sum-of-squares-of-special-elements/sum-of-squares-of-special-elements.java
class Solution {
    public int sumOfSquares(int[] nums) {
        int n=nums.length;
        int sum=0,l=nums[n-1];
        for(int i=1; i<=n/2; i++){
            if(n%i==0){
                int k=nums[i-1];
                sum+=k*k;
            }
        }
        return sum+(l*l);
    }
}

File: 2844-sum-of-squares-of-special-elements/sum-of-squares-of-special-elements.py
class Solution:
    def sumOfSquares(self, nums: List[int]) -> int:
        n=len(nums);s=0;l=nums[-1]
        for i in range(1,n//2+1):
            if n%i==0:
                k=nums[i-1]
                s+=k**2
        return s+(l**2)